sui/typescript/dapp_kit_example/src/client/App.tsx

```diff
@@ -0,0 +1,175 @@
+import "./App.css";
+import { useState } from "react";
+import { 
+  ConnectButton, 
+  useCurrentAccount,
+  useSignTransaction,
+  useSuiClient
+} from "@mysten/dapp-kit";
+import { Box, Flex, Heading } from "@radix-ui/themes";
+import axios from 'axios';
+import { SuiTransactionBlockResponse } from "@mysten/sui/client";
+import { useShinamiClient }  from "./hooks/useShinamiClient.js";
+
+function App() {
+  const currentAccount = useCurrentAccount();
+  const { mutateAsync: signTransaction } = useSignTransaction();
+  const client = useSuiClient();
+  const shinamiClient = useShinamiClient(); // My limited but functional replacement for dapp-kit client component and hook
```

Comment:
what was the thinking behind wrapping the suiClient in a context provider and creating the custom hook instead of using the useSuiClient hook directly?

Reply:
Just to play around and see what it might look like if we had first class support for frontend querying to our Node for React / dapp-kit apps. More requests to use = more $$. I see you said you'd take it out, which I'm fine with.  I'll see what Hao says.